examples/imm_example2.py

```diff
@@ -0,0 +1,287 @@
+import numpy as np
+from scipy.stats import multivariate_normal
+from matplotlib import pyplot as plt
+from copy import copy
+from datetime import datetime, timedelta
+
+
+from stonesoup.predictor.kalman import KalmanPredictor
+from stonesoup.updater.kalman import KalmanUpdater
+from stonesoup.predictor.mixturemodelproxy import add_mixture_capability
+
+from stonesoup.predictor.imm import IMMPredictor
+from stonesoup.updater.imm import IMMUpdater
+
+from stonesoup.models.transition.linear import ConstantVelocity, \
+    CombinedLinearGaussianTransitionModel, RandomWalk, OrnsteinUhlenbeck, \
+    ConstantAcceleration
+
+from stonesoup.models.measurement.linear import LinearGaussian
+
+from stonesoup.types.hypothesis import SingleHypothesis
+from stonesoup.types.detection import Detection
+from stonesoup.types.state import GaussianMixtureState, WeightedGaussianState
+from stonesoup.types.track import Track
+from stonesoup.types.prediction import GaussianStatePrediction, \
+    GaussianMeasurementPrediction,  MeasurementPrediction
+from stonesoup.types.array import StateVector, CovarianceMatrix
+
+from stonesoup.simulator.simple import SingleTargetGroundTruthSimulator
+
+from matplotlib.patches import Ellipse
+
+#######################################################################
+# This uses a CV and CA IMM. Inside the IMM uses the CA model
+# The taarget simulator uses a CV model.
+#######################################################################
+
+
+def cv2ca(state):
```

Comment:
I think that both cv2ca and ca2cv could be generalized to 2d and 3d and added to stonesoup.functions - they hare likely to be the mainstay for a lot of approaches like IMM.